File: TeamCode/src/main/java/org/hermitsocialclub/pandemicpanic/opmodes/ContourTestOp.java
package org.hermitsocialclub.pandemicpanic.opmodes;

import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
import com.qualcomm.robotcore.eventloop.opmode.TeleOp;
import org.hermitsocialclub.hydra.vision.DistanceToObjectDetector;
import org.hermitsocialclub.hydra.vision.VisionPipeline;
import org.hermitsocialclub.telecat.PersistantTelemetry;
import org.openftc.easyopencv.OpenCvCamera;
import org.openftc.easyopencv.OpenCvCameraFactory;
import org.openftc.easyopencv.OpenCvCameraRotation;
import org.openftc.easyopencv.OpenCvInternalCamera;

@TeleOp(name = "Contour Test")
public class ContourTestOp extends LinearOpMode {

    @Override
    public void runOpMode() {
        int cameraMonitorViewId = hardwareMap.appContext.getResources().getIdentifier("cameraMonitorViewId", "id", hardwareMap.appContext.getPackageName());
        OpenCvCamera phoneCam = OpenCvCameraFactory.getInstance().createInternalCamera(OpenCvInternalCamera.CameraDirection.BACK, cameraMonitorViewId);
        phoneCam.openCameraDevice();

        VisionPipeline pipeline = new VisionPipeline(new PersistantTelemetry(telemetry), new DistanceToObjectDetector());
        phoneCam.setPipeline(pipeline);
        phoneCam.startStreaming(320, 240, OpenCvCameraRotation.UPRIGHT);

        waitForStart();

        while (opModeIsActive()) {
            telemetry.addData("Frame Count", phoneCam.getFrameCount());
            telemetry.addData("FPS", String.format("%.2f", phoneCam.getFps()));
            telemetry.addData("Total frame time ms", phoneCam.getTotalFrameTimeMs());
            telemetry.addData("Pipeline time ms", phoneCam.getPipelineTimeMs());
            telemetry.addData("Overhead time ms", phoneCam.getOverheadTimeMs());
            telemetry.addData("Theoretical max FPS", phoneCam.getCurrentPipelineMaxFps());
            telemetry.update();

            if (gamepad1.dpad_up) {
                pipeline.setCannyUpperThreshold(pipeline.getCannyUpperThreshold() + 5);
            } else if (gamepad1.dpad_down) {
                pipeline.setCannyUpperThreshold(pipeline.getCannyUpperThreshold() - 5);
            }
            if (gamepad1.dpad_left) {
                pipeline.setCannyLowerThreshold(pipeline.getCannyLowerThreshold() + 5);
            } else if (gamepad1.dpad_right) {
                pipeline.setCannyLowerThreshold(pipeline.getCannyLowerThreshold() - 5);
            }

            sleep(10);
        }
    }

}

File: TeamCode/src/main/kotlin/org/hermitsocialclub/opmodes/ContourTestOp.kt
package org.hermitsocialclub.opmodes

import com.qualcomm.robotcore.eventloop.opmode.Disabled
import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import org.hermitsocialclub.hydra.vision.DistanceToObjectDetector
import org.hermitsocialclub.hydra.vision.VisionPipeline
import org.hermitsocialclub.telecat.PersistantTelemetry
import org.openftc.easyopencv.OpenCvCamera
@Disabled
@TeleOp(name = "Contour Test")
class ContourTestOp : AbstractVisionTestOp() {

    override fun buildPipeline(telemetry: PersistantTelemetry): VisionPipeline {
        return VisionPipeline(hardwareMap, telemetry, DistanceToObjectDetector())
    }

    override fun runLoop(telemetry: PersistantTelemetry, camera: OpenCvCamera, pipeline: VisionPipeline) {
        telemetry.setData("Frame Count", camera.frameCount)
        telemetry.setData("FPS", "%.2f", camera.fps)
        telemetry.setData("Canny Lower", pipeline.cannyLowerThreshold)
        telemetry.setData("Canny Upper", pipeline.cannyUpperThreshold)
        // telemetry.setData("Total frame time ms", phoneCam.totalFrameTimeMs)
        // telemetry.setData("Pipeline time ms", phoneCam.pipelineTimeMs)
        // telemetry.setData("Overhead time ms", phoneCam.overheadTimeMs)
        // telemetry.setData("Theoretical max FPS", phoneCam.currentPipelineMaxFps)

        if (gamepad1.dpad_up) {
            pipeline.cannyUpperThreshold = pipeline.cannyUpperThreshold + 5
        } else if (gamepad1.dpad_down) {
            pipeline.cannyUpperThreshold = pipeline.cannyUpperThreshold - 5
        }
        if (gamepad1.dpad_left) {
            pipeline.cannyLowerThreshold = pipeline.cannyLowerThreshold + 5
        } else if (gamepad1.dpad_right) {
            pipeline.cannyLowerThreshold = pipeline.cannyLowerThreshold - 5
        }

        sleep(10)
    }
}